src/components/VideoPlayer/index.js

```diff
@@ -100,10 +100,13 @@ export default class VideoPlayer extends PureComponent {
   }
 
   componentDidUpdate (prevProps) {
-    if (this.props.videoId !== prevProps.videoId
-      || this.props.hasAutoplay !== prevProps.hasAutoplay) {
+    if (this.props.videoId !== prevProps.videoId) {
       this.replaceWith(this.props.videoId)
     }
+
+    if (this.props.hasAutoplay !== prevProps.hasAutoplay) {
+      this.autoPlay()
+    }
```

Comment:
replace should only be called when you are in fact replacing the `videoId`, separating out the autoplay functionality if that has changed